File: src/test/java/brain/SchemaTest.java
package brain;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;
import java.util.Map;
import org.junit.jupiter.api.Test;

class SchemaTest {
  @Test
  void currentVersionIsOne() {
    assertEquals(1, Schema.CURRENT_SCHEMA_VERSION);
  }

  @Test
  void sensitivityOrdering() {
    assertTrue(Schema.level("public") < Schema.level("personal"));
    assertTrue(Schema.level("personal") < Schema.level("confidential"));
  }

  @Test
  void canCopyAllowsEqualOrLowerSource() {
    assertTrue(Schema.canCopy("personal", "confidential"));
    assertTrue(Schema.canCopy("personal", "personal"));
  }

  @Test
  void canCopyBlocksHigherSourceIntoLowerTarget() {
    assertFalse(Schema.canCopy("confidential", "personal"));
  }

  @Test
  void validateManifestAcceptsMinimalValid() {
    assertEquals(List.of(), Schema.validateManifest(Map.of("schema_version", 1, "area", "work")));
  }

  @Test
  void validateManifestReportsMissingArea() {
    assertTrue(Schema.validateManifest(Map.of("schema_version", 1)).stream()
        .anyMatch(e -> e.contains("area")));
  }

  @Test
  void validateManifestRejectsUnknownSensitivity() {
    assertTrue(Schema.validateManifest(
            Map.of("schema_version", 1, "area", "work", "sensitivity", "secret"))
        .stream().anyMatch(e -> e.contains("sensitivity")));
  }

  @Test
  void validateNoteRequiresDateAndSummary() {
    List<String> errors = Schema.validateNote(Map.of("schema_version", 1));
    assertTrue(errors.stream().anyMatch(e -> e.contains("date")));
    assertTrue(errors.stream().anyMatch(e -> e.contains("summary")));
  }

  @Test
  void validateNoteAcceptsValid() {
    assertEquals(List.of(),
        Schema.validateNote(Map.of("schema_version", 1, "date", "2026-05-31", "summary", "x")));
  }
}

File: test/schema.test.js
import test from "node:test";
import assert from "node:assert/strict";
import * as schema from "../brain_tools/schema.js";

test("current schema version is 1", () => {
  assert.equal(schema.CURRENT_SCHEMA_VERSION, 1);
});

test("sensitivity ordering", () => {
  assert.ok(schema.level("public") < schema.level("personal"));
  assert.ok(schema.level("personal") < schema.level("confidential"));
});

test("canCopy allows equal or lower source", () => {
  assert.equal(schema.canCopy("personal", "confidential"), true);
  assert.equal(schema.canCopy("personal", "personal"), true);
});

test("canCopy blocks higher source into lower target", () => {
  assert.equal(schema.canCopy("confidential", "personal"), false);
});

test("validateManifest accepts minimal valid", () => {
  assert.deepEqual(schema.validateManifest({ schema_version: 1, area: "work" }), []);
});

test("validateManifest reports missing area", () => {
  const errors = schema.validateManifest({ schema_version: 1 });
  assert.ok(errors.some((e) => e.includes("area")));
});

test("validateManifest rejects unknown sensitivity", () => {
  const errors = schema.validateManifest({ schema_version: 1, area: "work", sensitivity: "secret" });
  assert.ok(errors.some((e) => e.includes("sensitivity")));
});

test("validateNote requires date and summary", () => {
  const errors = schema.validateNote({ schema_version: 1 });
  assert.ok(errors.some((e) => e.includes("date")));
  assert.ok(errors.some((e) => e.includes("summary")));
});

test("validateNote accepts valid", () => {
  assert.deepEqual(schema.validateNote({ schema_version: 1, date: "2026-05-31", summary: "x" }), []);
});
